core/src/saros/net/stream/XMPPByteStreamAdapter.java

```diff
@@ -3,16 +3,30 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
+import java.util.Objects;
 import org.jivesoftware.smackx.bytestreams.BytestreamSession;
+import saros.net.xmpp.JID;
 
 public class XMPPByteStreamAdapter implements ByteStream {
 
   private final BytestreamSession delegate;
+  private final JID local;
+  private final JID remote;
+  private final String id;
+  private final StreamMode mode;
 
-  public XMPPByteStreamAdapter(final BytestreamSession session) {
-    if (session == null) throw new NullPointerException("session is null");
+  public XMPPByteStreamAdapter(
+      final JID localJid,
+      final JID remoteJid,
+      final BytestreamSession session,
+      final String id,
+      final StreamMode mode) {
 
-    delegate = session;
+    this.local = Objects.requireNonNull(localJid, "localJid is null");
+    this.remote = Objects.requireNonNull(remoteJid, "remoteJid is null");
+    this.delegate = Objects.requireNonNull(session, "session is null");
+    this.id = Objects.requireNonNull(id, "id is null");
+    this.mode = mode;
```

Comment:
getMode is returning NONE when mode is null.

Reply:
I know. Maybe my text could have been structured better.

The first question was why we allow null values for the mode at all. We have an alternative for cases where we don't have a mode (`NONE`), so just having the caller do the check and requiring the passed mode to be non-null would also be possible. Furthermore, this moves the responsibility to the caller, who may want to take additional steps in such a case.

The second question was why we hold the null reference. As the object is final, the null reference will also never change. And since you can access the held mode only though `getMode()` and it is never used internally otherwise, we can skip the check in `getMode()` by doing it once in the CTOR and then just holding the `NONE` mode if null is passed.